app/graphql/mutations/generate_access_token.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+module Mutations
+  class GenerateAccessToken < BaseMutation
+    argument :username, String, required: true
+    argument :password, String, required: true
+
+    field :access_token, String, null: true
+    field :user, Types::UserType, null: true
+
+    def execute(username:, password:)
+      @username = username
+      @password = password
+
+      authenticate_user
+      generate_token
+
+      {
+        access_token: @token,
+        user: @user
+      }
```

Comment:
I would've moved all of this into an interactor, since that's where business logic lives.